### DISCUSSION TITLE
Trying to Randomize Murderer Selection

### DISCUSSION THREAD

**Discussion by Beverly-Peders** [inkle/ink#811] [Writing in ink]
I'm creating a basic murder mystery game. I've set it up where players select a character and so each VAR character_name will equal the number of what order they were chosen in.

Example: 

VAR char_1 = 0
VAR char_2 = 0 
VAR character = 1 

=== character_selection ===
Select character: 
* char_1
 ~ char_1 += character
 ~ character += 1
  -> character_selection
* char_2
 ~ char_2 += character
 ~ character += 1
  -> character_selection
* continue -> setup 

Now I want to randomly select the character who is the murderer, but how do I do that when the information about the characters is in variable? I've gotten: 

VAR killer_selection = 0
~ killer_selection = RANDOM(1,character)

So it will have a number, but how can I make it search through the variables to select the one that has that number and name it the killer? 

Let me know if you need more info.

**Comment by Beverly-Peders** [comment]
Ah! I figured it out! Just needed a simple if bracket thing. It doesn't feel elegant, but this is what I have! 

{ 
    - desponia == killer_selection: 
        ~ killer += "desponia"
    - khione == killer_selection: 
        ~ killer += "khione"
    - ded_moroz == killer_selection: 
        ~ killer += "ded_moroz"
    - itzlacoliuhqui == killer_selection: 
        ~ killer += "itzlacoliuhqui"
    - marzanna == killer_selection: 
        ~ killer += "marzanna"
    - skadi == killer_selection: 
        ~ killer += "skadi"
    - hine_takurua == killer_selection: 
        ~ killer += "hine_takurua"
    - shakok == killer_selection: 
        ~ killer += "shakok"
    - else: 
        ~ killer += "naneh_sarma"
    
}

Now I have it so the killer has the name of the character. There's much more to do so I may be back if I get stumped again.

**Reply by Sklartacus** [reply]
Ah, thanks for this! I've been working on my own similar exercise that uses randomization in this way and this'll be helpful.
